cellrank/tl/kernels/_velocity_schemes.py

```diff
@@ -0,0 +1,345 @@
+# -*- coding: utf-8 -*-
+from abc import ABC, ABCMeta, abstractmethod
+from typing import Tuple
+from functools import partial
+
+import numpy as np
+from numba import njit
+
+from cellrank.ul._docs import d
+from cellrank.ul._utils import valuedispatch
+from cellrank.tl._constants import ModeEnum
+from cellrank.tl.kernels._utils import norm, np_mean, jit_kwargs
+
+
+class Scheme(ModeEnum):  # noqa: D101
+    DOT_PRODUCT = "dot_product"
+    COSINE = "cosine"
+    CORRELATION = "correlation"
+
+
+try:
+    import jax.numpy as jnp
+    from jax import jit, hessian
+
+    @jit
+    def _softmax_jax(x: np.ndarray, softmax_scale) -> np.ndarray:
+        numerator = x * softmax_scale
+        numerator = jnp.exp(numerator - jnp.max(numerator))
```

Comment:
Why is there the `-jnp.max(numerator)`?

Reply:
This is the softmax: https://en.wikipedia.org/wiki/Softmax_function

I really don't get this subtracted term, but realised that we had included it before. Why?

Reply:
It's for numerical stability, nothing else.

Reply:
Re stability:
```python3
import numpy as np
def _softmax(x: np.ndarray, softmax_scale: float) -> np.ndarray:
    numerator = x * softmax_scale
    numerator = np.exp(numerator)
    return numerator / np.sum(numerator)


def _softmax_stable(x: np.ndarray, softmax_scale: float) -> np.ndarray:
    numerator = x * softmax_scale
    numerator = np.exp(numerator - np.max(numerator))
    return numerator / np.sum(numerator)

np.random.seed(42)
_softmax(np.random.normal(size=(10,), loc=1000), 100)
# array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
np.random.seed(42)
_softmax_stable(np.random.normal(size=(10,), loc=1000), 100)
# array([1.68314704e-34, 1.33570495e-34, 7.25077912e-04, 1.81715293e-97,
       2.75338599e-89, 8.55578021e-39, 2.31953042e-58, 9.99274922e-01,
       8.26251375e-54, 1.03878654e-75])
```

Reply:
got it. Please add a comment - when working together on software, it's just much easier when these non-entirely obvious things are commented on. 
Definitely a good idea to use a stable version of the softmax here!